src/content/docs/en/guides/upgrade-to/v5.mdx

```diff
@@ -662,6 +662,66 @@ Note that `src/env.d.ts` is only necessary if you have added custom configuratio
 
 <ReadMore>Read more about [TypeScript configuration in Astro](/en/guides/typescript/#setup).</ReadMore>
 
+### Changed: Actions submitted by HTML forms no longer use cookies
+
+In Astro 4.x, actions called from an HTML form would trigger a redirect with the result forwarded using cookies. This caused issues for large form errors and return values that exceeded the 4 KB limit of cookie-based storage.
+
+Astro 5.0 now renders the result of an action as a POST result without any forwarding. This reintroduces the "confirm form resubmission?" dialog when a user attempts to refresh the page, and will not persist action results across sessions.
+
+If you need to customize this behavior, we've introduced the `getActionContext()` utility to handle action results from middleware however you choose. You can reimplement the cookie forwarding behavior of Astro 4.X, or implement your own session storage to avoid the 4 KB size limit and persist results across sessions. 
+
+<ReadMore>See the [advanced sessions guide](/en/guides/actions/#advanced-persist-action-results-with-a-session) for a sample implementation using Netlify Blob.</ReadMore>
+
+#### What should I do?
+
+You can use the new default without any changes to your code. If you accept the 4 KB size limit and prefer the cookie forwarding behavior from 4.X, you can implement redirects using middleware as shown in this sample snippet:
```

Comment:
```suggestion
#### What should I do?

You can use the new default without any changes to your code. If you accept the 4 KB size limit and prefer the cookie forwarding behavior from 4.X, you can implement redirects using a new `getActionContext()` utility to handle action results from middleware:
```
It didn't really feel like the above fit before the "what should I do?"

Reply:
Ah, I see what you mean. I included that because I _want_ people to read the session tutorial first before blindly copy / pasting the cookies example. We definitely want to steer people towards sessions first to avoid the pitfalls of cookies. Perhaps we can say that we recommend using session-based storage, but here's an sample to implement the 4.X behavior if session storage is prohibitive for you

Reply:
OK, what about a structure like:


> You can use the new default without any changes to your code. **Session-based storage, our new default described in docs is totally way better.**
>
> If you accept the 4 KB size limit and **insist on being unreasonable and want to use the cookie forwarding behavior from 4.X instead of the recommended session-based storage**, you can implement redirects using a new `getActionContext()` utility to handle action results from middleware:

So, bold is where we can emphasize (not those words, but that's where we can nudge the reader). Is that anything?